src/tests/system/tests/test_netgroups.py

```diff
@@ -108,3 +109,145 @@ def test_netgroups__add_remove_netgroup_member(client: Client, provider: Generic
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(testhost5, ng2, ldap.test)" in member
+    assert "(testhost1, ng1, ldap.test)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_netgroup__lookup_nested_groups(client: Client, provider: GenericProvider):
+    """
+    :title: Nesting netgroups and verifying user memberships using LDAP with sssd.
+    :setup:
+        1. Create users, groups.
+        2. Create netgroup named netgroup and Add Member
+        3. Create another netgroup named nested_netgroup
+        4. Add Members to nested_netgroup
+        5. Add Circular Netgroup Nesting to nested_netgroup
+        6. Start sssd
+    :steps:
+        1. Retrieves all members of the "nested_netgroup" group using the getent netgroup tool.
+        2. Verify that users from another group is also part of "nested_netgroup".
+        3. Checks if a user who is not in any netgroup is part of "nested_netgroup".
+        4. After the SSSD restart, it retrieves the members of "nested_netgroup" again to ensure they still intact.
+    :expectedresults:
+        1. All members of the "nested_netgroup" group be there
+        2. Users from another group is also part of "nested_netgroup".
+        3. User who is not in any netgroup is part of "nested_netgroup".
+        4. After restart all members of the "nested_netgroup" group be there
+    """
+    if isinstance(provider, IPA):
+        pytest.skip(reason="Not for IPA povider")
+
+    for id in [1, 2, 3]:
+        provider.user(f"ng{id}").add()
+
+    netgroup = provider.netgroup("netgroup").add()
+    netgroup.add_member(host="testhost1", user="ng1", domain="ldap.test")
+
+    nested_netgroup = provider.netgroup("nested_netgroup").add()
+    nested_netgroup.add_member(ng=netgroup.dn)
+    nested_netgroup.add_member(host="testhost5", user="ng2", domain="ldap.test")
+    nested_netgroup.add_member(user="ng3")
+
+    netgroup.add_member(ng=nested_netgroup.dn)
+
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("nested_netgroup").members
+    assert "(testhost1,ng1,ldap.test)" in member
+    assert "(-,ng3,)" in member
+    assert "(testhost5,ng2,ldap.test)" in member
+
+    client.sssd.restart()
+
+    member = client.tools.getent.netgroup("nested_netgroup").members
+    assert "(testhost1,ng1,ldap.test)" in member
+    assert "(-,ng3,)" in member
+    assert "(testhost5,ng2,ldap.test)" in member
+
+
+@pytest.mark.parametrize(
+    "user, domain, expected",
+    [("host", "host.ldap.test", "(host,-,host.ldap.test)"), ("ng3", "", "(-,ng3,)")],
+)
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_netgroup__host_and_domain(client: Client, provider: GenericProvider, user: str, domain: str, expected: str):
+    """
+    :title: Netgroup contains a member that only has a host and domain specified, but no associated user.
```

Comment:
With the parametrization you are actually defining a user as a netgroup member. Consider updating the comment to match the test behaviour.